swift-sdk/Internal/InAppPersistence.swift

```diff
@@ -0,0 +1,175 @@
+        case consumed
+    }
+    
+    public convenience init(from decoder: Decoder) {
+        guard let container = try? decoder.container(keyedBy: CodingKeys.self) else {
+            ITBError("Can not decode, returning default")
+            self.init(messageId: "",
+                      campaignId: "",
+                      content: IterableHtmlInAppContent(edgeInsets: .zero, backgroundAlpha: 0.0, html: "")
+                      )
+            return
+        }
+        
+        guard let contentType = (try? container.decode(IterableInAppContentType.self, forKey: .contentType)), contentType == .html else {
+            // unexpected content type
+            ITBError("Unexpected contentType, returning default")
+            self.init(messageId: "",
+                      campaignId: "",
+                      content: IterableHtmlInAppContent(edgeInsets: .zero, backgroundAlpha: 0.0, html: "")
+            )
+            return
+        }
+        
+        let messageId = (try? container.decode(String.self, forKey: .messageId)) ?? ""
+        let campaignId = (try? container.decode(String.self, forKey: .campaignId)) ?? ""
+        let channelName = (try? container.decode(String.self, forKey: .channelName)) ?? ""
+        let content = (try? container.decode(IterableHtmlInAppContent.self, forKey: .content)) ?? IterableHtmlInAppContent(edgeInsets: .zero, backgroundAlpha: 0.0, html: "")
+        let extraInfo = (try? container.decode([String : String].self, forKey: .extraInfo)) ?? nil
+        
+        self.init(messageId: messageId,
+                  campaignId: campaignId,
+                  channelName: channelName,
+                  contentType: contentType,
+                  content: content,
+                  extraInfo: extraInfo)
+        
+        self.processed = (try? container.decode(Bool.self, forKey: .processed)) ?? false
+        self.consumed = (try? container.decode(Bool.self, forKey: .consumed)) ?? false
+    }
+    
+    public func encode(to encoder: Encoder) {
+        guard let content = content as? IterableHtmlInAppContent else {
+            return
+        }
+        
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try? container.encode(messageId, forKey: .messageId)
+        try? container.encode(campaignId, forKey: .campaignId)
+        try? container.encode(channelName, forKey: .channelName)
+        try? container.encode(contentType, forKey: .contentType)
+        try? container.encode(content, forKey: .content)
+        try? container.encode(extraInfo, forKey: .extraInfo)
+        try? container.encode(processed, forKey: .processed)
+        try? container.encode(consumed, forKey: .consumed)
+    }
+}
+
+protocol InAppPersistenceProtocol {
+    func getMessages() -> [IterableInAppMessage]
+    func persist(_ messages: [IterableInAppMessage])
+    func clear()
+}
+
+
+class FilePersister : InAppPersistenceProtocol {
+    init(filename: String = "itbl", ext: String = "json") {
```

Comment:
We may need to store multiple files for other things in the future, what do you think about adding inapp to the name? Like `itbl_inapp.json` or similar.